src/borg/testsuite/repository.py

```diff
@@ -15,11 +15,29 @@
 from ..remote import RemoteRepository, InvalidRPCMethod, PathNotAllowed
 from ..repository import Repository, LoggedIO, MAGIC, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
 from ..repoobj import RepoObj
-from . import BaseTestCase
 from .hashindex import H
 
 
-UNSPECIFIED = object()  # for default values where we can't use None
+@pytest.fixture()
+def repository(tmpdir):
+    # setup
+    repository_location = os.path.join(str(tmpdir), "repository")
+    repository = Repository(repository_location, exclusive=True, create=True)
+    yield repository.__enter__()
+
+    # teardown
+    repository.__exit__(None, None, None)
+    # shutil.rmtree(str(tmpdir))
```

Comment:
This line caused the windows tests to fail on last commit, need to figure out why. Commented it out for now.

Reply:
check the tmpdir fixture docs - i think tmpdir does cleanup itself anyway.

Reply:
btw, guess some try/finally is missing here.